src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -211,42 +187,311 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
       jobSetId = jobSetId,
       namespace = conf.get(ARMADA_SPARK_JOB_NAMESPACE),
-      priority = conf.get(ARMADA_SPARK_JOB_PRIORITY),
+      priority = Some(conf.get(ARMADA_SPARK_JOB_PRIORITY)),
       containerImage = containerImage,
       podLabels = podLabels,
       driverLabels = driverLabels,
       executorLabels = executorLabels,
       nodeSelectors = nodeSelectors.getOrElse(Map.empty),
       nodeUniformityLabel = gangUniformityLabel,
-      armadaClusterUrl = armadaClusterUrl,
-      executorConnectionTimeout = Duration(conf.get(ARMADA_EXECUTOR_CONNECTION_TIMEOUT), SECONDS)
+      armadaClusterUrl = Some(armadaClusterUrl),
+      executorConnectionTimeout =
+        Some(Duration(conf.get(ARMADA_EXECUTOR_CONNECTION_TIMEOUT), SECONDS)),
+      driverLimitCores = driverLimitCores,
+      driverRequestCores = driverRequestCores,
+      driverLimitMemory = driverLimitMemory,
+      driverRequestMemory = driverRequestMemory,
+      executorLimitCores = executorLimitCores,
+      executorRequestCores = executorRequestCores,
+      executorLimitMemory = executorLimitMemory,
+      executorRequestMemory = executorRequestMemory
     )
   }
 
-  private[spark] case class ArmadaJobConfig(
-      queue: String,
-      jobSetId: String,
+  /** Resolves a value based on precedence: CLI > Template > Default
+    *
+    * @param cliValue
+    *   Value from CLI configuration
+    * @param templateValue
+    *   Value from template
+    * @param defaultValue
+    *   Default value to use if neither CLI nor template provide a value
+    * @return
+    *   Resolved value based on precedence
+    */
+  private def resolveValue[T](
+      cliValue: Option[T],
+      templateValue: => Option[T],
+      defaultValue: => T
+  ): T = {
+    cliValue.orElse(templateValue).getOrElse(defaultValue)
+  }
+
+  /** Resolves a required value based on precedence: CLI > Template > Error
+    *
+    * @param cliValue
+    *   Value from CLI configuration
+    * @param templateValue
+    *   Value from template
+    * @param errorMessage
+    *   Error message if no value is found
+    * @return
+    *   Resolved value based on precedence
+    */
+  private def resolveRequiredValue[T](
+      cliValue: Option[T],
+      templateValue: => Option[T],
+      errorMessage: => String
+  ): T = {
+    cliValue.orElse(templateValue).getOrElse(throw new IllegalArgumentException(errorMessage))
+  }
+
+  /** Helper class to resolve common configuration values using precedence */
+  private case class ResolvedConfig(
       namespace: String,
       priority: Double,
       containerImage: String,
+      armadaClusterUrl: String,
+      executorConnectionTimeout: Duration
+  )
+
+  /** Resolves all required configuration values using precedence */
+  private def resolveConfig(cliConfig: CLIConfig, conf: SparkConf): ResolvedConfig = {
+    val namespace = resolveValue(cliConfig.namespace, None, "default")
+    val priority  = resolveValue(cliConfig.priority, None, 0.0)
```

Comment:
The defaults for namespace and priority are already in Config.scala, (or should be.)  Can we avoid hard-coding them here?